public/app/features/alerting/unified/components/rule-editor/GrafanaEvaluationBehavior.tsx

```diff
@@ -181,51 +153,42 @@ export function GrafanaEvaluationBehaviorStep({
     control,
   } = useFormContext<RuleFormValues>();
 
-  const [folder, group, type, isPaused, folderUid, folderName] = watch([
-    'folder',
+  const [group, type, isPaused, folderUid, folderName, evaluateEvery] = watch([
     'group',
     'type',
     'isPaused',
     'folder.uid',
     'folder.title',
+    'evaluateEvery',
   ]);
 
   const isGrafanaAlertingRule = isGrafanaAlertingRuleByType(type);
   const isGrafanaRecordingRule = isGrafanaRecordingRuleByType(type);
-  const { groupOptions, loading } = useFolderGroupOptions(folder?.uid ?? '', enableProvisionedGroups);
+  const { currentData: rulerNamespace, isLoading: loadingGroups } = useFetchGroupsForFolder(folderUid ?? '');
   const [isEditingGroup, setIsEditingGroup] = useState(false);
 
-  const rulerRuleRequests = useUnifiedAlertingSelector((state) => state.rulerRules);
-  const groupfoldersForGrafana = rulerRuleRequests[GRAFANA_RULES_SOURCE_NAME];
+  const groupOptions = useMemo(() => {
+    return rulerNamespace ? namespaceToGroupOptions(rulerNamespace, enableProvisionedGroups) : [];
+  }, [enableProvisionedGroups, rulerNamespace]);
 
-  const grafanaNamespaces = useCombinedRuleNamespaces(GRAFANA_RULES_SOURCE_NAME);
-  const existingNamespace = grafanaNamespaces.find((ns) => ns.uid === folderUid);
-  const existingGroup = existingNamespace?.groups.find((g) => g.name === group);
+  const existingGroup = Object.values(rulerNamespace ?? {})
+    .flat()
+    .find((ruleGroup) => ruleGroup.name === group);
+  const existingNamespaceName = Object.keys(rulerNamespace ?? {}).at(0);
 
-  const isNewGroup = useIsNewGroup(folderUid ?? '', group);
+  const isNewGroup = !existingGroup && !loadingGroups;
 
+  // synchronize the evaluation interval with the group name when it's an existing group
   useEffect(() => {
-    if (!isNewGroup && existingGroup?.interval) {
-      setEvaluateEvery(existingGroup.interval);
+    if (existingGroup) {
+      setValue('evaluateEvery', existingGroup.interval ?? DEFAULT_GROUP_EVALUATION_INTERVAL);
```

Comment:
is this necessary?

Reply:
Yes, when we select a different group from the dropdown we need to update the interval of the form to make sure the form values are kept in sync with what the user has selected.

Ideally we'd store a "evaluation group" which contains both the name and the interval but that seemed like a much larger refactor.

Reply:
But I think this is for the existing group, not the selected one? (dependencies are on the `existingGroup`)